### PR TITLE
Code Review - Kerry Healey (healeyke)

### PR DESCRIPTION
**Pull request by healeyke**
**Analysis**
Overall, I feel like this project was missing quite a few important files/classes.  One of the main issues I ran into while reviewing the project is that there was no readme file.  This made reviewing the project challenging because I didn't know what the project was called, what it was supposed to do, what different classes the project needed, etc.  I went ahead and created the missing basic files that needed to be added: ReadMe, RetrofitClientInstance, AppModule, ComputerPartsApplication, MainViewModel, ui/theme, and its corresponding files.  Also, I made quite a few edits to the actual code of the project, like I added imports for the Retrofit and Dispatchers so there was no error in the ComponentService file (which was commented out, don't do this in your final points will be taken off).  In general, there was not much in this project for me to go off of code-wise, so I didn't really know how to grade.

**Was the program available in UC Github on time?**
- Was it made available before the assignment due date, yes.  But I believe that the group had some trouble with their original repository so they created a new one, but I didn't find out about it until this past weekend a few days before the assignment was due.

**Is the program documented/commented well enough for you to understand?**
- There was really no documentation that I could find within the project.  After completing the code review, I did however find a DesignDoc file that is somewhat like a ReadMe, but it lacked key information on the project and was missing important aspects (class diagram, functional requirements, etc.)

**Does the program compile?**
- I believe it compiles.

**Rationale behind changes.**
- For most of my changes to the project, I was just adding classes and files that were missing.  Some of the key ones I add (like I said above) were a ReadMe, RetrofitClientInstance, AppModule, ComputerPartsApplication, MainViewModel, ui/theme, and its corresponding files.  
- Also, I made a few changes to the areas of the code I could, like in the ComponentService file, all of the code was commented out because there were two red lines within it.  I fixed both of these errors by adding the RetrofitClientInstance, and importing both Retrofit and Dispatchers into the file and all errors were resolved.
- There wasn't much else that I could comment on or change because there wasn't much to go off of.

**Links to three specific commits that I made to your own group's GitHub repository (Cookit!) before the end of sprint deadline.**  
- [add new test branch](https://github.com/healeyke/Cookit-/commit/9e9b7ab94b9ac4c7e58e29b1e95cf879494183de)
- [added config.yml from lecture](https://github.com/healeyke/Cookit-/commit/390acd6d80c7040c5636c3b827038b483b1b1d3a)
- [add Chicken Alfredo to IntegrationTest](https://github.com/healeyke/Cookit-/commit/892ef58c16ec9e4b32880d4a13ea32cc62efe974) 

**Technical Concepts**
- Importance of code organization and thorough documentation (makes reviewing code a lot easier)
- Understand why certain classes and files were needed in order for other classes and files to function properly (example: ComponentService needed Retrofit)
- Importance of specific naming conventions in testing and for variables, methods, classes, etc. (makes code more readable and understandable to viewers who didn't contribute to the project)

**Sources used for code review:**
- https://kotlinlang.org/docs/reference/coding-conventions.html
- https://developer.android.com/distribute/best-practice
- https://docs.sonarqube.org/display/SONARQUBE45/Developers%27+Seven+Deadly+Sins

### CODE REVIEW

**Comment by healeyke** [comment]
I added my specific code comments within my commits

File: app/src/main/java/com/example/springprojectstationbuild/AppModule.kt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.example.springprojectstationbuild
+
+class AppModule {
+}
```
Comment (discospiff) [inline app/src/main/java/com/example/springprojectstationbuild/AppModule.kt:4]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

File: app/src/main/java/com/example/springprojectstationbuild/ComputerPartsApplication.kt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package com.example.springprojectstationbuild
+
+class ComputerPartsApplication {
+}
```
Comment (discospiff) [inline app/src/main/java/com/example/springprojectstationbuild/ComputerPartsApplication.kt:4]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

File: app/src/main/java/com/example/springprojectstationbuild/MainViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package com.example.springprojectstationbuild
+
+class MainViewModel {
+}
```
Comment (discospiff) [inline app/src/main/java/com/example/springprojectstationbuild/MainViewModel.kt:4]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

File: app/src/main/java/com/example/springprojectstationbuild/RetrofitClientInstance.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.example.springprojectstationbuild
+
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+object RetrofitClientInstance {
+    private var retrofit: Retrofit? = null
+    private val BASE_URL = ""
```
Comment (discospiff) [inline app/src/main/java/com/example/springprojectstationbuild/RetrofitClientInstance.kt:8]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

**Comment by discospiff** [comment]
The additions in this branch are likely needed for the project at some point, but are new, and sometimes incomplete, functionality.  New functionality is better in a separate branch until fully developed, as features should not be merged to main/master until the meet the team's Definition of Done.  Code review branches should minimize technical debt, and thus, be merge-able to master immediately.  So, if the team desires these features, I'd keep the branch open, and iterate over the feature until it is complete.

**Pull request closed without merging**